src/components/Welcome.jsx

```diff
@@ -1,16 +1,28 @@
 import React, { Component } from 'react';
 import { Link } from "react-router-dom";
+import queryString from 'query-string';
 
 class Welcome extends Component {
+
+    componentDidMount() {
+        const query = queryString.parse(location.search);
+        const data = JSON.parse(query.info || '{}');
+        localStorage.setItem('token', data.token);        
+    }
+
     render() {
+
         return (
-        <div>
-            <h1>You are welcome at Barefoot nomad</h1>
-			<Link to='/login'><h1>Click me to login</h1></Link>
-            <Link to='/redux'><h1>Click check redux</h1></Link>
-		</div>
+            <div>
+                <h1>You are welcome at Barefoot nomad</h1>
+                <Link to='/login'><h1>Click me to login</h1></Link>
+                <Link to='/auth'><h1>Click me to login with social accounts</h1></Link>
```

Comment:
am wondering why you change /auth to be a main route of application. my suggestion is that we can stick on /   empty route and redirect user who is not already signed in to login page and those who signed in and already have a valid token to the dashboard.

Reply:
@Jajabenit250 No, it is not the home link, that is the link for navigating to the social sign-in buttons. we still redirect the user to the homepage if he is already signed in.

Reply:
@jo-rdan  Jordan my point is if user is already signed in and we have token in local storage that are valid can he use your route again? I was thinking maybe remove everything from local storage at the beginning of your component because I can see in your implementation you are setting token to local storage but in case I access the route again I can still set token while there is a token which already exists in local storage. or check if token in local storage is valid and then redirect him to dashboard if not valid give him access to using signing in using social auth.

Reply:
@Jajabenit250 I understand your concern but you are working on the sign-in feature, I think that you are in charge of that because if you really think about it my feature is just another part of your feature, my buttons will appear in your overall component, I worry about the functionality of the social authentication

Reply:
@jo-rdan okay I get that. that's why I was suggesting if posting change that /auth route to / route. because /auth is no use to your feature. if you are only working on functionality.